src/main/java/org/eclipse/tractusx/managedidentitywallets/service/DidDocumentService.java

```diff
@@ -42,8 +45,20 @@ public class DidDocumentService {
      * @param identifier the identifier
      * @return the did document
      */
-    public DidDocument getDidDocument(String identifier) {
-        return commonService.getWalletByIdentifier(identifier).getDidDocument();
+    public DidDocument getDidDocument(String didOrBpn) {
+        if (commonService.checkIfDid(didOrBpn)) {
+            return commonService.getWalletByDid(new Identifier(didOrBpn)).getDidDocument();
```

Comment:
here again. Previously identifier was referring to did or BPN. Now it is referring to DID